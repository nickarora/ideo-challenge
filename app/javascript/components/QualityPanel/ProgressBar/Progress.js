import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'

export const Bar = styled.div`
  height: 100%;
`

const Progress = ({ progress, color }) => (
  <Bar
    style={{
      background: color,
      width: `${Math.abs(progress)}%`,
    }}
  />
)

Progress.propTypes = {
  progress: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
}

export default Progress
