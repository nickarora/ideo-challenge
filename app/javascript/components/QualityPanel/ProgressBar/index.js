import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'
import Progress from './Progress'

export const Container = styled.section`
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 24px;
  margin-bottom: 24px;
`

export const NegativeProgress = styled.div`
  display: flex;
  justify-content: flex-end;
  flex: 1;
  height: 6px;
`

export const Divider = styled.div`
  width: 1px;
  border: 1px black dashed;
  height: 100%;
`

export const PositiveProgress = styled.div`
  display: flex;
  justify-content: flex-start;
  flex: 1;
  height: 6px;
`

const ProgressBar = ({ quality: { score, color } }) => (
  <Container>
    <NegativeProgress>
      {score < 0 && <Progress progress={score} color={color} />}
    </NegativeProgress>
    <Divider />
    <PositiveProgress>
      {score > 0 && <Progress progress={score} color={color} />}
    </PositiveProgress>
  </Container>
)

ProgressBar.propTypes = {
  quality: PropTypes.shape({
    color: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
  }).isRequired,
}

export default ProgressBar
