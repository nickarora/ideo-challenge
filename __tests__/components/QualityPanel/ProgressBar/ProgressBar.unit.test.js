import ProgressBar from '~/components/QualityPanel/ProgressBar'

describe('ProgressBar', () => {
  it('renders with a positive score', () => {
    const wrapper = shallow(
      <ProgressBar
        quality={{
          score: 50,
          color: '#333',
        }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with a negative score', () => {
    const wrapper = shallow(
      <ProgressBar
        quality={{
          score: -50,
          color: '#333',
        }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with a score of zero', () => {
    const wrapper = shallow(
      <ProgressBar
        quality={{
          score: -50,
          color: '#333',
        }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
