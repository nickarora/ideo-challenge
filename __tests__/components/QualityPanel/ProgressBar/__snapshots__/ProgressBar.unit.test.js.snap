// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders with a negative score 1`] = `
<styled.section>
  <styled.div>
    <Progress
      color="#333"
      progress={-50}
    />
  </styled.div>
  <styled.div />
  <styled.div />
</styled.section>
`;

exports[`ProgressBar renders with a positive score 1`] = `
<styled.section>
  <styled.div />
  <styled.div />
  <styled.div>
    <Progress
      color="#333"
      progress={50}
    />
  </styled.div>
</styled.section>
`;

exports[`ProgressBar renders with a score of zero 1`] = `
<styled.section>
  <styled.div>
    <Progress
      color="#333"
      progress={-50}
    />
  </styled.div>
  <styled.div />
  <styled.div />
</styled.section>
`;
